### PR TITLE
Component type generation

### LINKED ISSUE #46: [TODO] automatically generate component types

### PR DESCRIPTION
**Pull request by betaboon**
resolves #46

### CODE REVIEW

File: components/navigation-drawer/src/NavigationDrawer.svelte [hunk 1]
```diff
@@ -1,9 +1,9 @@
 <script lang="ts">
-  // render NavigationDrawer in open state
+  /** whether NavigationDrawer is open */
   export let open = false;
-  // render NavigationDrawer as modal
+  /** whether NavigationDrawer is modal */
   export let modal = false;
-  // enable elevation (does not apply to modal)
+  /** wheter NavigationDrawer is elevated */
```
Comment (Florian-Schoenherr) [inline components/navigation-drawer/src/NavigationDrawer.svelte:6]: whether 🙂

File: tools/svelte-actions/src/ripple.ts [hunk 2]
```diff
@@ -24,14 +32,14 @@ class Ripple {
...
-  private animationElement: HTMLElement;
+  private animationElement: HTMLElement | undefined = undefined;
 
   constructor(element: HTMLElement, options: RippleOptions) {
     this.element = element;
     this.options = { ...RippleDefaults, ...options };
     this.injectContainerStyle();
     if (this.options.triggerOnParent) {
-      this.triggerElement = this.element.parentElement;
+      this.triggerElement = this.element.parentElement || this.element;
```
Comment (Florian-Schoenherr) [inline tools/svelte-actions/src/ripple.ts:42]: Where does this occur? Seems a little counter-intuitive if someone uses `triggerOnParent` and it still triggers on the element. Maybe we could warn or just error or `if` there?

**Review by Florian-Schoenherr — CHANGES_REQUESTED** [review]
a big, biiig major problem and a question :)

File: tools/svelte-actions/src/ripple.ts (hunk 2 above)
Comment (betaboon) [reply tools/svelte-actions/src/ripple.ts:42]: this should only happen if an element has no parent. which is ... `body` ? or even just `html` ?
the main reason for adding it was making typescript happy :P

**Pull request merged**